src/basic/chattr-util.c

```diff
@@ -145,6 +145,11 @@ int read_attr_fd(int fd, unsigned *ret) {
         if (!S_ISDIR(st.st_mode) && !S_ISREG(st.st_mode))
                 return -ENOTTY;
 
+        _cleanup_close_ int fd_close = -EBADF;
+        fd = fd_reopen_condition(fd, O_RDONLY|O_CLOEXEC|O_NOCTTY, O_PATH, &fd_close); /* drop O_PATH if it is set */
```

Comment:
Hmm? fd and fd_close usage got seriously messed up here?

Reply:
why that? fd_reopen_condition() is supposed to this helper that fixes up the open flags of an fd if they don't match expectations, but only then. Thus:

1. it always returns the fd to ultimately use as ret val
2. if it needs to reopen the fd for this because the flags don't match expectations then it returns that new fd in the last parameter, but only then, and otherwise it returns EBADF there. That last param must be closed always.

i added the helper precisely to make functions like this work that call syscalls that don#t support O_PATH right now, but really should, and hence we work around it.

Yes, they are not the most obvious calling semantics, but I think they do make sense.